src/routes/movies.py

```
@@ -370,117 +781,202 @@
 
-    :raises HTTPException: Raises a 404 error if the movie with the given ID is not found.
+    return new_genre
 
-    :return: A response indicating the successful update of the movie.
-    :rtype: None
+
+@router.put("genres/{genre_id}", tags=["Genres"], response_model=GenreSchema)
+async def update_genre(
+        genre_id: int,
+        new_name: str,
+        db: AsyncSession = Depends(get_db)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)